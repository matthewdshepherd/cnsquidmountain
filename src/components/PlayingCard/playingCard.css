.playing--card__div {
    display: flex;
    justify-content: center;
    align-items: center;
    align-content: center;
    border-radius: 10px;
    border: 2px solid black;
    width: 80%;
    background-color: #FFF;
    margin-bottom: 2%;
    height: 15%;
}

.selected--card {
    border: 2px solid blue;
    box-shadow: 0 4px 8px 0 #505050, 0 6px 20px 1px #505050;
}

.incorrect--card {
    border: 2px solid red;
}

.playing--card__div:hover {
    cursor: pointer;
    box-shadow: 0 4px 8px 0 #505050, 0 6px 20px 1px #505050;
}

.playing--card__p--text {
    color: black;
    font-size: 45px;
    font-weight: 700;
}