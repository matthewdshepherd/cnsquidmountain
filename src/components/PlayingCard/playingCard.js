import React from 'react'
import './playingCard.css'

let classNames = require('classnames');

export const PlayingCard = ({ prefix, handleChange, value, column, completedWords, incorrect, currentCorrect }) => {
  let playingCardClass = classNames({
    'playing--card__div': !completedWords.includes(parseInt(prefix.id)),
    'hide--card__div': completedWords.includes(parseInt(prefix.id)),
    'selected--card': parseInt(column) === parseInt(prefix.id),
    'incorrect--card': parseInt(incorrect) === parseInt(prefix.id),
    'correct--card': parseInt(currentCorrect) === parseInt(prefix.id)
  })

  return (
        <div  className={playingCardClass} data-id={prefix.id} data-value={value} key={prefix.id} onClick={handleChange}>
            <p className="playing--card__p--text" data-value={value} data-id={prefix.id} >{prefix.prefix}</p>
        </div>
  )
}
export default PlayingCard;