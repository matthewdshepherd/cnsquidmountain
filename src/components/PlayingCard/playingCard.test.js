import React from 'react';
import { shallow } from 'enzyme';
import PlayingCard from './playingCard';

describe('PlayingCard', () => {
    it('Should match the snapshot with all data passed in correctly', () => {
        const wrapper = shallow(<PlayingCard key={1} prefix={'e/ex'} handleChange={jest.fn()} value={'column1'}/> )
        expect(wrapper).toMatchSnapshot();
    })
})