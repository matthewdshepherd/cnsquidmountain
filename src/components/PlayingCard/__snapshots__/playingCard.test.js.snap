// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayingCard Should match the snapshot with all data passed in correctly 1`] = `
<div
  className="playing--card__div column1"
  data-value="column1"
  onClick={[MockFunction]}
  type="button"
>
  <p
    className="playing--card__p--text"
    data-value="column1"
  />
</div>
`;
